pymc3/variational/updates.py

```diff
@@ -0,0 +1,150 @@
+###########################################################
+#  Refactored by Dilin Wang
+#  Original implementation by Alec Radford, Luke Metz, Soumith Chintala: https://github.com/Newmu/dcgan_code   
+#  (c) Alec Radford, Luke Metz, Soumith Chintala
+###########################################################
+
+import theano
+import theano.tensor as tt
+import numpy as np
+
+
+__all__ = [
+    "SGD",
+    "Momentum",
+    "Adagrad",
+    "RMSprop",
+    "Adadelta",
+    "Adam"
+]
+
+def floatX(X):
+    return np.asarray(X, dtype=theano.config.floatX)
+
+class Update(object):
+
+    def __init__(self):
+        self.__dict__.update(locals())
+
+    def __call__(self, params, grads):
```

Comment:
@twiecki I think **Lasagne** has good and reliable implementation. If they don't mind copy-paste that would be great. Otherwise we should probably change api so that it will be the same as Lasagne's api.